### PR TITLE
Add suffixes to SDK Windows commands

### PR DESCRIPTION
**Pull request by awholegunch**
Fixes a bug that prevented the CDAP `tx-debugger.bat` from starting under Windows.

Added code to both `cdap-cli` and `tx-debugger` to identify themselves in their usage statements correctly as to the correct suffix on the commands.

Built SDK, tested on Mac and Windows. (And re-tested with latest version.)

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java [hunk 1]
```diff
@@ -335,6 +335,14 @@ private static void addOptionalOption(Options options, Option option) {
   }
 
   private static void usage() {
+    String toolName;
+    String toolNameBase = "cdap-cli";
+    String os = System.getProperty("os.name").toLowerCase();
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java]: Can you use the `OSDetector` class in cdap-common module?

Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java]:
```java
String toolName = "cdap-cli" + OSDetector.isWindows() ? ".bat" : ".sh";
```

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java [hunk 2]
```diff
@@ -229,9 +229,18 @@ private void printUsage(boolean error) {
...
+    String toolName;
+    String os = System.getProperty("os.name").toLowerCase();
+    if (os.indexOf("win") >= 0) {
+      toolName = TOOL_NAME_BASE + ".bat";
+    } else {
+      toolName = TOOL_NAME_BASE + ".sh";
+    }
+    
     pw.println("Usage:" +
-        "\n\t " + TOOL_NAME + " view [ <option> ... ]" +
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: how did it work previously, without the script suffix? Just left to the user to append the suffix as they desire?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java [hunk 3]
```diff
@@ -65,8 +65,8 @@
   private static final Gson GSON = new Gson();
   private static final SimpleDateFormat formatter
       = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S z");
-  private static final String TOOL_NAME = "tx-debugger";
-
+  private static final String TOOL_NAME_BASE = "tx-debugger";
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]:
```java
private static final String toolName = "tx-debugger" + OSDetector.isWindows() ? ".bat" : ".sh";
```

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java (hunk 2 above)
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: Well, there wasn't any desire involved. If you wanted it to work, you had to use the correct suffix...

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java (hunk 3 above)
Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: Fixed.

File: cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java (hunk 1 above)
Comment (awholegunch) [inline cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java]: Fixed.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java [hunk 4]
```diff
@@ -229,9 +230,11 @@ private void printUsage(boolean error) {
     } else {
       pw = new PrintWriter(System.out);
     }
+    
+    private static final String toolName = "tx-debugger" + OSDetector.isWindows() ? ".bat" : ".sh";
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: sorry, don't need to use `private static final` prefix, when its used from a method.
Just if it was declared as a field of the class.

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: in fact... how does this even compile?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java (hunk 3 above)
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: unused `TOOL_NAME_BASE`?

Comment (awholegunch) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: Actually this is wrong. It needs to be:
```
String toolName = "tx-debugger" + (OSDetector.isWindows() ? ".bat" : ".sh");

File: cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java (hunk 1 above)
Comment (awholegunch) [inline cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java]: Though this too needs to be (precedence, man, precedence):
```
String toolName = "cdap-cli" + (OSDetector.isWindows() ? ".bat" : ".sh");

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java (hunk 3 above)
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/TransactionManagerDebuggerMain.java]: this constant is unused. The line (69) can probably be removed.

**Comment by anwar6953** [comment]
lgtm

**Pull request merged**